File: scripts/common_py/keyring_helper.py
import keyring

from scripts.common_py.GUOHUB_CLI_CONFIG import NAME_SPACE


def guohub_get_credential(name):
    return keyring.get_password(NAME_SPACE, name)


def guohub_set_credential(name, value):
    keyring.set_password(NAME_SPACE, name, value)


def guohub_delete_credential(name):
    keyring.delete_password(NAME_SPACE, name)

File: scripts/common_js/keyring_helper.ts
import { setPassword, getPassword, deletePassword } from "cross-keychain";
import { NAME_SPACE } from "./GUOHUB_CLI_CONFIG.js";

export async function getCredential(name: string) {
  return await getPassword(NAME_SPACE, name);
}

export async function setCredential(name: string, value: string) {
  await setPassword(NAME_SPACE, name, value);
}

export async function deleteCredential(name: string) {
  await deletePassword(NAME_SPACE, name);
}
